src/main/java/frc/robot/subsystems/led/LedMatrixSubsystem.java

```diff
@@ -0,0 +1,139 @@
+        if (self){
+        for (int i=0; i<colors.length; i += 2) {
+            Collections.reverse(Arrays.asList(colors[i]));
+        };
+        self = false;
+    }
+        for (int i=0; i<colors.length; i++) { 
+            for (int j=0; j<colors[i].length; j++){
+                buffer.setLED((16*i)+j, hex2Rgb(colors[i][j]));
+            }
+        }
+        leds.setData(buffer);
+    }
+    boolean self2 = true;
+    public void setLedOne(){
+        if (self2){
+            for (int i=0; i<colors.length; i += 2) {
+                Collections.reverse(Arrays.asList(one[i]));
+            };
+            self2 = false;
+        }
+            for (int i=0; i<colors.length; i++) { 
+                for (int j=0; j<colors[i].length; j++){
+                    buffer.setLED((16*i)+j, hex2Rgb(one[i][j]));
+                }
+            }
+            leds.setData(buffer);
+    }
+    boolean self3 = true;
+    public void setLedTwo(){
+        if (self3){
+            for (int i=0; i<colors.length; i += 2) {
+                Collections.reverse(Arrays.asList(two[i]));
+            };
+            self3 = false;
+        }
+            for (int i=0; i<colors.length; i++) { 
+                for (int j=0; j<colors[i].length; j++){
+                    buffer.setLED((16*i)+j, hex2Rgb(two[i][j]));
+                }
+            }
+            leds.setData(buffer);
+    }
+    boolean self4 = true;
+    public void setLedThree(){
+        if (self4){
+            for (int i=0; i<colors.length; i += 2) {
+                Collections.reverse(Arrays.asList(three[i]));
+            };
+            self4 = false;
+        }
+            for (int i=0; i<colors.length; i++) { 
+                for (int j=0; j<colors[i].length; j++){
+                    buffer.setLED((16*i)+j, hex2Rgb(three[i][j]));
+                }
+            }
+            leds.setData(buffer);
+    }
+    boolean self5 = true;
+    public void setLedGoCans(){
+        if (self5){
+            for (int i=0; i<colors.length; i += 2) {
+                Collections.reverse(Arrays.asList(gocans[i]));
+            };
+            self5 = false;
+        }
+            for (int i=0; i<colors.length; i++) { 
+                for (int j=0; j<colors[i].length; j++){
+                    buffer.setLED((16*i)+j, hex2Rgb(gocans[i][j]));
+                }
+            }
+            leds.setData(buffer);
+    }
+
+    int instance = 0;
+    public void setRainbowLed(){
+        double bright = ledBrightnessSlider.getDouble(0.2);
+        for (int i = 0; i < 256; i++) {
+            if (i%2==0 && instance%3==0){
+              buffer.setHSV(i, (instance + i)/100, 255, (int) (255 * bright)); //Change the divisor in the hue to change the fade speed
+            } else {
+                buffer.setHSV(i, 0, 0, 0);
+            }
+            instance = instance + 1;
+        }
+        leds.setData(buffer);
+    }
+
+
+
+    @Override
+    public void periodic() {
+        setRainbowLed();
```

Comment:
Make into a command if you want this capability on the robot. Remove it from periodic either way